src/main/src/main/java/org/geoserver/security/decorators/SecuredFeatureSource.java

```diff
@@ -89,12 +89,14 @@ protected SecuredFeatureSource(FeatureSource<T, F> delegate, WrapperPolicy polic
                     ReTypingFeatureCollection retyped = new ReTypingFeatureCollection(sfc, target);
                     return (FeatureCollection) SecuredObjects.secure(retyped, policy);
                 } else {
-                    // complex feature store eh? No way to fix it at least warn the admin
-                    LOGGER.log(
-                            Level.SEVERE,
-                            "Complex store returned more properties than allowed "
-                                    + "by security (because they are required by the schema). "
-                                    + "Either the security setup is broken or you have a security breach");
+                    if (!readQuery.equals(Query.ALL)) {
```

Comment:
Ah ok, so this hides the log in case the security subsystem specifies no restrictions right?
However, the check seems to be too broad, what if the security subsystem was returning a filter to reduce the features, while keeping all the attributes? Better to check the readQuery for the property list, and see if all properties were allowed instead.